app/Http/Controllers/ShoppingCartController.php

```diff
@@ -4,12 +4,21 @@
 
 use App\Models\Product;
 use App\Models\Species;
-use Illuminate\Http\Request;
+use Illuminate\Http\Request;    
+use Illuminate\Http\RedirectResponse;
+use Illuminate\Http\JsonResponse;
+use Illuminate\View\View;
 
 class ShoppingCartController extends Controller
 {
-    public function add(Request $request, $id)
+    public function add(Request $request, int $id): RedirectResponse
     {
+        $request->validate([
```

Comment:
Don't forget to ask Daniel about this, where do we put this verification or do we do it with Javascript? @Juanstevan1